backtrace-library/src/androidTest/java/backtraceio/library/breadcrumbs/BacktraceBreadcrumbsTest.java

```diff
@@ -0,0 +1,635 @@
+                assertEquals("I am a breadcrumb", parsedBreadcrumb.get("message"));
+                assertNotNull(parsedBreadcrumb.get("From_Thread"));
+                assertEquals("manual", parsedBreadcrumb.get("type"));
+                assertEquals("info", parsedBreadcrumb.get("level"));
+                // Timestamp should be convertible to a number
+                Long.parseLong(parsedBreadcrumb.get("timestamp"));
+                // Id should be convertible to a number
+                Integer.parseInt(parsedBreadcrumb.get("id"));
+            }
+
+            Comparator<Map<String, String>> mapComparator = new Comparator<Map<String, String>>() {
+                public int compare(Map<String, String> m1, Map<String, String> m2) {
+                    return Integer.parseInt(m1.get("id")) - Integer.parseInt(m2.get("id"));
+                }
+            };
+
+            Collections.sort(parsedBreadcrumbList, mapComparator);
+
+        } catch(Exception ex) {
+            fail(ex.getMessage());
+        }
+    }
+
+    @Test
+    public void testQueueFileRolloverCustomMax() {
+        int numIterations = 100;
+
+        try {
+            BacktraceBreadcrumbs backtraceBreadcrumbs = new BacktraceBreadcrumbs(context, 6400);
+            backtraceBreadcrumbs.enableBreadcrumbs();
+
+            for (int i = 0; i < numIterations; i++) {
+                final long threadId = Thread.currentThread().getId();
+                Map<String, Object> attributes = new HashMap<String, Object>() {{
+                    put("From Thread", threadId);
+                }};
+                backtraceBreadcrumbs.addBreadcrumb("I am a breadcrumb", attributes);
+            }
+
+            List<String> breadcrumbLogFileData = readBreadcrumbLogFiles(backtraceBreadcrumbs);
+
+            List<Map<String, String>> parsedBreadcrumbList = new ArrayList<Map<String, String>>();
+            // We should have rolled over the configuration breadcrumb, consider all breadcrumbs here
+            for (int i = 0; i < breadcrumbLogFileData.size(); i++)
+            {
+                Map<String, String> parsedBreadcrumb = parseBreadcrumb(breadcrumbLogFileData.get(i));
+                parsedBreadcrumbList.add(parsedBreadcrumb);
+                assertEquals("I am a breadcrumb", parsedBreadcrumb.get("message"));
+                assertNotNull(parsedBreadcrumb.get("From_Thread"));
+                assertEquals("manual", parsedBreadcrumb.get("type"));
+                assertEquals("info", parsedBreadcrumb.get("level"));
+                // Timestamp should be convertible to a number
+                Long.parseLong(parsedBreadcrumb.get("timestamp"));
+                // Id should be convertible to a number, and should skew larger (earlier breadcrumbs rolled off)
+                assertTrue(Integer.parseInt(parsedBreadcrumb.get("id")) > 45);
+            }
+
+            Comparator<Map<String, String>> mapComparator = new Comparator<Map<String, String>>() {
```

Comment:
Try to extract this 

```
      Comparator<Map<String, String>> mapComparator = new Comparator<Map<String, String>>() {
                public int compare(Map<String, String> m1, Map<String, String> m2) {
                    return Integer.parseInt(m1.get("id")) - Integer.parseInt(m2.get("id"));
                }
            };
``` because it is duplicated a few times

Reply:
Will do!